File: frontend/src/api/publish.ts
import { http, type ApiResponse } from './request'

export interface PublishRecord {
  id: number
  user_id: number
  content_id: number
  status: number // 0:待发布, 1:发布中, 2:成功, 3:失败
  error_msg?: string
  scheduled_at: string
  published_at?: string
  created_at: string
  updated_at: string
  content?: any
}

export interface SchedulePublishRequest {
  content_id: number
  publish_time: string // RFC3339格式
  frequency?: string // once, daily, weekly
}

export interface PublishNowRequest {
  content_id: number
}

export interface PublishListResponse {
  list: PublishRecord[]
  total: number
  page: number
  page_size: number
}

/**
 * 立即发布内容
 */
export const publishNow = async (data: PublishNowRequest): Promise<ApiResponse<PublishRecord>> => {
  return await http.post('/publish/now', data)
}

/**
 * 定时发布内容
 */
export const schedulePublish = async (data: SchedulePublishRequest): Promise<ApiResponse<PublishRecord>> => {
  return await http.post('/publish/schedule', data)
}

/**
 * 获取发布记录列表
 */
export const getPublishRecords = async (params: {
  page?: number
  page_size?: number
}): Promise<ApiResponse<PublishListResponse>> => {
  return await http.get('/publish/list', { params })
}

/**
 * 获取发布记录详情
 */
export const getPublishRecord = async (id: number): Promise<ApiResponse<PublishRecord>> => {
  return await http.get(`/publish/${id}`)
}

/**
 * 取消发布
 */
export const cancelPublish = async (id: number): Promise<ApiResponse<any>> => {
  return await http.post(`/publish/${id}/cancel`)
}

/**
 * 重试发布
 */
export const retryPublish = async (id: number): Promise<ApiResponse<PublishRecord>> => {
  return await http.post(`/publish/${id}/retry`)
}

File: backend/internal/model/publish.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// PublishRecord 发布记录模型
type PublishRecord struct {
	ID         uint           `json:"id" gorm:"primaryKey;comment:发布记录ID"`
	UserID     uint           `json:"user_id" gorm:"index;not null;comment:用户ID"`
	ContentID  uint           `json:"content_id" gorm:"index;not null;comment:内容ID"`
	Status     int            `json:"status" gorm:"default:0;comment:状态 0:待发布 1:发布中 2:成功 3:失败"`
	ErrorMsg   string         `json:"error_msg" gorm:"type:text;comment:错误信息"`
	ScheduledAt time.Time     `json:"scheduled_at";comment:计划发布时间"`
	PublishedAt *time.Time    `json:"published_at";comment:实际发布时间"`
	CreatedAt  time.Time      `json:"created_at"`
	UpdatedAt  time.Time      `json:"updated_at"`
	DeletedAt  gorm.DeletedAt `json:"-" gorm:"index"`
	User       User           `json:"user,omitempty" gorm:"foreignKey:UserID"`
	Content    Content        `json:"content,omitempty" gorm:"foreignKey:ContentID"`
}

// TableName 指定表名
func (PublishRecord) TableName() string {
	return "publish_records"
}

// SchedulePublishRequest 定时发布请求
type SchedulePublishRequest struct {
	ContentID   uint   `json:"content_id" binding:"required"`
	PublishTime string `json:"publish_time" binding:"required"`
	Frequency   string `json:"frequency"`
}

// PublishRequest 立即发布请求
type PublishRequest struct {
	ContentID uint `json:"content_id" binding:"required"`
}
